app/models/concerns/foreman_ansible/host_managed_extensions.rb

```diff
@@ -43,14 +42,20 @@ def play_ansible_roles
                     "#{e.message}")
       end
       # rubocop:enable Metrics/AbcSize
+
+      def all_ansible_roles
+        (ansible_roles + inherited_ansible_roles).uniq
+      end
     end
     # rubocop:enable Metrics/BlockLength
     # Class methods we may need to override or add
     module ClassMethods
-      def import_host(hostname, certname = nil, deprecated_proxy = nil)
-        host = super(hostname, certname, deprecated_proxy)
-        if IPAddress.valid?(hostname) && Nic::Interface.find_by(:ip => hostname)
-          host = Nic::Interface.find_by(:ip => hostname).host
+      def import_host(*args)
```

Comment:
Does this break compatibility with any version? As far as I can see previous versions should be able to work with it too, but if you could confirm it it'd be great

Reply:
It rather improves compatibility. :) We simply pass on arguments to `super` without touching and later get the first argument as hostname.

I tested this with the definition of `import_host` before https://github.com/theforeman/foreman/pull/5855 and it works as well.